goth_tests/domain/ya-provider/test_provider_break_agreement.py

```diff
@@ -0,0 +1,260 @@
+"""End to end tests for requesting WASM tasks using goth REST API clients."""
+
+import logging
+from pathlib import Path
+from typing import List
+
+import pytest
+
+from goth.address import (
+    PROXY_HOST,
+    YAGNA_REST_URL,
+)
+from goth.configuration import load_yaml, Override
+from goth.node import node_environment
+from goth.runner import Runner
+from goth.runner.container.payment import PaymentIdPool
+from goth.runner.container.yagna import YagnaContainerConfig
+from goth.runner.probe import RequestorProbe, ProviderProbe
+
+from goth_tests.helpers.activity import run_activity, wasi_exe_script, wasi_task_package
+from goth_tests.helpers.negotiation import negotiate_agreements, DemandBuilder
+from goth_tests.helpers.payment import pay_all
+
+logger = logging.getLogger(__name__)
+
+
+def _topology(assets_path: Path) -> List[YagnaContainerConfig]:
```

Comment:
We're moving away from defining the topology in code. This can cause all sort of problems for you, especially when mixing configuration modes (i.e. using a topology and a config file in one test).
One example of an issue is CLI config overrides won't work properly (e.g. to use local `yagna` binaries instead of default ones).

Instead, you should use either the default config with some overrides (you can take a look at e.g. `test_zero_amount_txs.py`) or create a new `goth-config.yml` based on the default one.

If you need to set environment variables we have a PR which adds support for this in `goth-config.yml`: https://github.com/golemfactory/goth/pull/505

Reply:
The config feature I mentioned is now merged to `goth` master, I'm going to make a release including that change today .

Reply:
Great. I'm working on this, but we have feature freeze and I have to postpone it.